spec/spec_helper.rb

```diff
@@ -6,6 +6,7 @@
 require 'factory_girl'
 require 'faker'
 require 'vcr'
+require 'pry'
```

Comment:
I would not add to the git repo, the `require pry` to the spec helper. I am not sure others, but if I need to debug I added the require and binding pry altogether when I need debugging (but do not commit).

Reply:
If you manually do this all the time why not just do it once and save time and be helpful in the future? Its added as a dev dependancy so should not effect anything in the gem.

Reply:
The pry on the Gemfile is fine. I would be honest that I don't debug all the time, it is rarely that I do, usually when the code is legacy. Debugging indicates that the code is not simple to understand what's going on and could be an indication the need of refactoring it. Again the pry on the Gemfile is fine, requiring it I don't think so as I explained above.

Reply:
Sorry but this has nothing to do with how often and when you should be debugging. This is adding a `require` so that if you _want_ to add a `binding.pry` to debug something, you can.

What are you worried about by adding this code? Are you worried that by adding a require for a debugger - it makes the code _appear_ legacy? or more complex?

We can have a separate discussion about whether or not you should need a debugger when writing software but it seems ridiculous to block this?

Reply:
I don't have a very strong opinion either way, but I can agree with @tomas-stefano reasoning. The question I would ask when reviewing a line like this would be "is this piece of code necessary for the application or the tests to run" - the answer being no. When I debug, I put the code in that I need to find the answers and then will remove the unnecessary code again. I also for example sometimes will choose `byebug` over `pry`. 

Leaving code in the source just to aid debugging feels like a code smell to me. the argument that it saves time doesn't feel that valid considering how little time it takes to write in a require when you do occasionally debug. 

The focus of @tomas-stefano original comment I think is more that it doesn't keep the code clean